cpp/CMakeLists.txt

```diff
@@ -585,6 +585,7 @@ target_include_directories(
 target_compile_definitions(
   cudf PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUDF_CXX_DEFINITIONS}>"
               "$<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:CUDA>:${CUDF_CUDA_DEFINITIONS}>>"
+       PRIVATE "THRUST_WRAPPED_NAMESPACE=rapids"
```

Comment:
This currently is PRIVATE on purpose since making it public would break consumers that use Thrust and don't call cudf API that have thrust types.